src/Media/Ingester/SideloadDir.php

```diff
@@ -0,0 +1,409 @@
+<?php declare(strict_types=1);
+
+namespace FileSideload\Media\Ingester;
+
+use Omeka\Api\Request;
+use Omeka\Entity\Media;
+use Omeka\File\TempFileFactory;
+use Omeka\File\Validator;
+use Omeka\Media\Ingester\IngesterInterface;
+use Omeka\Stdlib\ErrorStore;
+use Omeka\Stdlib\Message;
+use Laminas\Form\Element;
+use Laminas\View\Renderer\PhpRenderer;
+
+/**
+ * @todo Factorize with Sideload and avoid to list files and directory in main directory twice.
+ */
+class SideloadDir implements IngesterInterface
+{
+    /**
+     * @var string
+     */
+    protected $directory;
+
+    /**
+     * @var bool
+     */
+    protected $deleteFile;
+
+    /**
+     * @var int
+     */
+    protected $maxFiles;
+
+    /**
+     * @var TempFileFactory
+     */
+    protected $tempFileFactory;
+
+    /**
+     * @var Validator
+     */
+    protected $validator;
+
+    /**
+     * @var array
+     */
+    protected $listDirs = [];
+
+    /**
+     * @var bool
+     */
+    protected $hasMoreDirs = false;
+
+    /**
+     * @param string $directory
+     * @param bool $deleteFile
+     * @param int $maxFiles
+     * @param TempFileFactory $tempFileFactory
+     * @param Validator $validator
+     */
+    public function __construct(
+        $directory,
+        $deleteFile,
+        $maxFiles,
+        TempFileFactory $tempFileFactory,
+        Validator $validator
+    ) {
+        // Only work on the resolved real directory path.
+        $this->directory = $directory ? realpath($directory) : '';
+        $this->deleteFile = $deleteFile;
+        $this->maxFiles = $maxFiles;
```

Comment:
Please move `$maxFiles` to the end of the signature, like you did with the Sideload ingester.

Reply:
Fixed in last commit.